realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -299,6 +318,12 @@ private boolean categorizeClassElements() {
                         return false;
                     }
 
+                    // A field cannot be nullable and primary key. Currently only String can be nullable
+                    if (Utils.isString(variableElement)
+                            && nullableFieldNames.contains(variableElement)) {
+                        nullableFieldNames.remove(variableElement);
```

Comment:
Strings are by default nullable, and if I find a primary key, that particular field cannot be nullable. Therefore it is removed from the set of nullable fields.

If I process the annotation in the opposite order, I will have to check every string field if it is a primary key.